### DISCUSSION TITLE
docker部署后，TimeoutError: ResourceRequest timed out

### DISCUSSION THREAD

**Discussion by Andy-njupt** [walinejs/waline#1821] [Q&A]
使用Vercel部署正常，docker部署后，获取发布comment等操作，提示TimeoutError: ResourceRequest timed out；
以下是我的配置：
     docker拉取的mongo镜像，version5.0，配置的环境变量：MONGO_DB=admin；MONGO_USER=waline_test；MONGO_PASSWORD=xxx；MONGO_HOST=127.0.0.1；MONGO_PORT=27017。在admin数据库上，waline_test的角色是userAdminAnyDataBase;
    docker拉取的lizheming/waline，配置的环境变量同上；
    打开本地服务http://127.0.0.1:8360，页面展示正常；前端配置之后也可以正常连通这个服务；但是涉及到调用这个服务接口的操作会报以下错误：
TimeoutError: ResourceRequest timed out
at ResourceRequest._fireTimeout (/app/node_modules/generic-pool/lib/ResourceRequest.js:62:17)
at Timeout.bound (/app/node_modules/generic-pool/lib/ResourceRequest.js:8:15)
at listOnTimeout (node:internal/timers:564:17)
at processTimers (node:internal/timers:507:7)

我不清楚我配置的过程中，哪个环节出现了问题，还请帮忙看以下，感谢！！！

**Comment by hdwdhdw** [comment]
您好，我也遇到了同样的问题，请问您解决了吗？

---
Hello, I also encountered the same problem, have you solved it?

**Reply by lizheming** [reply]
Docker 部署可以换成 SQLite 数据库，这个错误信息应该是 MongoDB 配置不太正确的原因，MongoDB 的配置格式有点奇怪，需要多试一下，有点麻烦。

---
Docker deployment can be replaced by a SQLite database. This error message should be because the MongoDB configuration is not correct. The configuration format of MongoDB is a bit strange. It needs to be tried more, which is a bit troublesome.

**Comment by yangyang5214** [comment]
> 参数配置的问题

https://waline.js.org/guide/database.html#mongodb

这里的环境变量去试 

- 失败

<img width="1256" height="368" alt="image" src="https://github.com/user-attachments/assets/f8a6aee0-bc73-4a26-86f7-85033c087f1f" />

- 正常

<img width="1158" height="267" alt="image" src="https://github.com/user-attachments/assets/155eb1c4-4165-4350-99fb-358c65bcfeee" />
